src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -504,7 +505,10 @@ def _process_wrapper(
     log_context_initialization_func,
     operation_contexts_dict: dict,
 ):
-    signal.signal(signal.SIGINT, signal_handler)
+    if threading.current_thread() is threading.main_thread():
+        signal.signal(signal.SIGINT, signal_handler)
```

Comment:
why there is no signal.signal(signal.SIGTERM, signal_handler) in batch_engine.py ? they could not be terminated by kill ?

Reply:
Let's discuss it in a separate thread. @PeiwenGaoMS